general/hash.hpp

```diff
@@ -212,6 +213,84 @@ class HashTable : public BlockArray<T>
 };
 
 
+/// Hash function for data sequences.
+/** Depends on GnuTLS for SHA-256 hashing. */
+class HashFunction
```

Comment:
Should this be `#ifdef MFEM_USE_GNUTLS`?

Reply:
No need -- the implementation has the `#ifdef`s. If you don't have GnuTLS, you'll get a hash string that says you need GnuTLS.

Reply:
Ok, thanks. Do you want to mention any of the hash changes in `CHANGELOG`?

Reply:
It's not crucial.

Reply:
Is GnuTLS already installed on LC, or does someone know how to install it? Is it the one at https://github.com/gnutls/gnutls so we just need to clone and follow their build instructions?

Reply:
It is already installed on LC in `/usr/lib64`, so during config you just need to add `MFEM_USE_GNUTLS=YES`.

Reply:
Thanks @v-dobrev, I tried that but apparently need something else:
In file included from general/hash.cpp(15):
/usr/include/gnutls/crypto.h(35): error: identifier "gnutls_cipher_algorithm_t" is undefined
                       gnutls_cipher_algorithm_t cipher,

Reply:
On Mac I got it working, using a `brew` installation of `gnutls`. Maybe on LC some include directory needs to be specified? Anyway, the hash feature is a really nice addition.

Reply:
I get the same error with @dylan-copeland. All paths look ok... so It's not clear to me what's wrong here

Reply:
Thanks for testing this. I pushed a fix for the issue -- that version of GnuTLS needed `<gnutls/gnutls.h>` to be included before `<gnutls/crypto.h>`.

Reply:
Thanks, I confirmed it works now on LC.